modules/args.py

```diff
@@ -27,4 +27,8 @@ def get_args():
         required=True,
         help="the location to stream downloaded files to, i.e. rtmp://localhost/stream/live"
     )
+    parser.add_argument(
+        "--cache-file",
+        help="file to store the cache state",
+    )
```

Comment:
have the user specify the full path to the cache state file. then we dont need to check if the file ends with `.json`
```suggestion
    parser.add_argument(
        "--cache-state-file",
        help="JSON file to persist cache state on server shutdown and recover on startup. if specified, the server will not empty the cache on shutdown"
    )
```

Reply:
why do we not check if the file ends with .json when the user specifies the full path? The user could specify a path to text file or other files instead of json, so imo, we still have to check that

Reply:
the user should be able to specify whatever path they want, its simple to just use whatever they pass in, and write json to it

if the user specifies a file like `/tmp/state.txt`, the server should try to parse/write json to it anyway

the name of the file doesnt affect anything besides how a human reads the name